tutorials/blitz/1_introduction_by_pytorch_lightning.rst

```diff
@@ -0,0 +1,508 @@
+    # 
+    # This tutorial will build a two-layer `Graph Convolutional Network
+    # (GCN) <http://tkipf.github.io/graph-convolutional-networks/>`__. Each
+    # layer computes new node representations by aggregating neighbor
+    # information.
+    # 
+    # To build a multi-layer GCN you can simply stack ``dgl.nn.GraphConv``
+    # modules, which inherit ``torch.nn.Module``.
+    # 
+
+.. code:: ipython3
+
+    from dgl.nn import GraphConv
+    from torch.optim.lr_scheduler import ReduceLROnPlateau
+    
+    class GCN(pl.LightningModule):
+        def __init__(self, in_feats, h_feats, num_classes):
+            super(GCN, self).__init__()
+            self.conv1 = GraphConv(in_feats, h_feats)
+            self.conv2 = GraphConv(h_feats, num_classes)
+            self.train_accuracy = torchmetrics.Accuracy()
+            self.val_accuracy = torchmetrics.Accuracy()
+            self.test_accuracy = torchmetrics.Accuracy()
+        
+        def forward(self, g, in_feat):
+            h = self.conv1(g, in_feat)
+            h = F.relu(h)
+            h = self.conv2(g, h)
+            return h
+        def training_step(self,batch,batch_idx):
+          g = batch
+          features = g.ndata['feat']
+          labels = g.ndata['label']
+          train_mask = g.ndata['train_mask']
+          val_mask = g.ndata['val_mask']
+          test_mask = g.ndata['test_mask']
+          # Forward
+          logits = self(g, features)
+    
+          # Compute prediction
+          pred = logits.argmax(1)
+    
+          # Compute loss
+          # Note that you should only compute the losses of the nodes in the training set.
+          loss = F.cross_entropy(logits[train_mask], labels[train_mask])
+    
+          
+          # Compute accuracy on training/validation/test
+          self.train_accuracy(pred[train_mask],labels[train_mask])
+          self.val_accuracy(pred[val_mask],labels[val_mask])
+          self.test_accuracy(pred[test_mask],labels[test_mask])
+          self.log_dict({'train/loss':loss,'train/accuracy':self.train_accuracy},prog_bar=True,on_step=True,on_epoch=True,batch_size=1) # set the batch_size =1
+          return loss
+    
+        def validation_step(self,batch,batch_idx):
+            self.log_dict({'valid/accuracy':self.val_accuracy,'test/accuracy':self.test_accuracy},prog_bar=True,on_step=True,on_epoch=True,batch_size=1) 
+    
+        def predict_step(self,batch,batch_idx):
+          g = batch
+          features = g.ndata['feat']
+          # Forward
+          logits = self(g, features)
+          return logits
+    
+        def configure_optimizers(self):
+            optimizer = torch.optim.Adam(self.parameters(),lr=1e-2,weight_decay=1e-7)
+            lr_scheduler = {
+                "scheduler": ReduceLROnPlateau(
+                    optimizer,
+                    mode='min',
+                    factor=0.9,
+                    patience=15,
+                    min_lr=1e-3
+                ),
```

Comment:
We don't have separate closing parenthesis/braces/brackets/etc.  You can put it after the last element.